lib/pacemaker/pcmk_result_code.c

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2022 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <crm/cib/internal.h>
+#include <crm/crm.h>
+
+#include <pacemaker.h>
+#include <pacemaker-internal.h>
+
+/*!
+ * \internal
+ * \brief Get the name and description of a result code
+ *
+ * \param[in]  code  The result code
+ * \param[in]  type  Interpret \c code as this type of result code. Supported
+ *                   values: \c pcmk_result_legacy, \c pcmk_result_pcmk_rc,
+ *                   \c pcmk_result_crm_exitcode.
+ * \param[out] name  Where to store the result code's name
+ * \param[out] desc  Where to store the result code's description
+ *
+ * \return Standard Pacemaker return code
+ *
+ * \note This function should return \c pcmk_rc_ok even if the result code is
+ *       unknown. It will return an error only if the lookup function returns
+ *       \c NULL, which should never happen.
+ */
+static int
+get_strings(int code, pcmk_result_type_t type, const char **name,
+            const char **desc)
+{
+    CRM_ASSERT((type == pcmk_result_legacy) || (type == pcmk_result_pcmk_rc)
+               || (type == pcmk_result_crm_exitcode));
```

Comment:
It can sometimes be confusing, but you could also write this like:
```
CRM_ASSERT(type <= pcmk_result_crm_exitcode);
```

Reply:
I probably will, just because there's a precedent. It's already used quite a lot within Pacemaker.

To be honest, that convention always gives me a headache, because understanding it requires going back and checking the order of the enum. An enum is technically ordered but it's usually fairly arbitrary, like here -- `pcmk_result_ocf_exitcode` is not greater than `pcmk_result_crm_exitcode` in any semantically meaningful way.

Reply:
Didn't see this. I'll revert and update.